kunta-api-server/src/main/java/fi/otavanopisto/kuntaapi/server/integrations/mwp/MwpNewsProvider.java

```diff
@@ -0,0 +1,333 @@
+    return Collections.emptyList();
+  }
+
+  @Override
+  public NewsArticle findOrganizationNewsArticle(OrganizationId organizationId, NewsArticleId newsArticleId) {
+    Post post = findPostByArticleId(newsArticleId);
+    if (post != null) {
+      return translateNewsArticle(post);
+    }
+
+    return null;
+  }
+
+  @Override
+  public List<Attachment> listNewsArticleImages(OrganizationId organizationId, NewsArticleId newsArticleId) {
+    Post post = findPostByArticleId(newsArticleId);
+    if (post != null) {
+      Integer featuredMediaId = post.getFeaturedMedia();
+      if (featuredMediaId != null) {
+        fi.otavanopisto.mwp.client.model.Attachment featuredMedia = findMedia(featuredMediaId);
+        if ((featuredMedia != null) && (featuredMedia.getMediaType() == MediaTypeEnum.IMAGE)) {
+          return Collections.singletonList(translateAttachment(featuredMedia));
+        }
+      }
+    }
+
+    return Collections.emptyList();
+  }
+
+  @Override
+  public Attachment findNewsArticleImage(OrganizationId organizationId, NewsArticleId newsArticleId,
+      AttachmentId attachmentId) {
+    
+    Post post = findPostByArticleId(newsArticleId);
+    if (post != null) {
+      Integer featuredMediaId = post.getFeaturedMedia();
+      if (featuredMediaId != null) {
+        AttachmentId mwpAttachmentId = getImageAttachmentId(featuredMediaId);
+        if (!idController.idsEqual(attachmentId, mwpAttachmentId)) {
+          return null;
+        }
+        
+        fi.otavanopisto.mwp.client.model.Attachment attachment = findMedia(featuredMediaId);
+        if (attachment != null) {
+          return translateAttachment(attachment);
+        }
+      }
+    }
+
+    return null;
+  }
+
+  @Override
+  public AttachmentData getNewsArticleImageData(OrganizationId organizationId, NewsArticleId newsArticleId,
+      AttachmentId attachmentId, Integer size) {
+    
+    Integer mediaId = getMediaId(attachmentId);
+    if (mediaId == null) {
+      return null;
+    }
+    
+    fi.otavanopisto.mwp.client.model.Attachment featuredMedia = findMedia(mediaId);
+    if (featuredMedia.getMediaType() == MediaTypeEnum.IMAGE) {
+      AttachmentData imageData = getImageData(featuredMedia.getSourceUrl());
+      
+      if (size != null) {
+        return scaleImage(imageData, size);
+      } else {
+        return imageData;
+      }
+      
+    }
+    
+    return null;
+  }
+  
+  private AttachmentData scaleImage(AttachmentData imageData, Integer size) {
+    BufferedImage bufferedImage = imageReader.readBufferedImage(imageData.getData());
+    if (bufferedImage != null) {
+      BufferedImage scaledImage = imageScaler.scaleMaxSize(bufferedImage, size);
+      byte[] scaledImageData = imageWriter.writeBufferedImageAsPng(scaledImage);
+      if (scaledImageData != null) {
+        return new AttachmentData("image/png", scaledImageData);
+      }
+    }
+    
+    return null;
+  }
+  
+  private AttachmentData getImageData(String imageUrl) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "getImageData" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)